python/cugraph/cugraph/tests/test_node2vec.py

```diff
@@ -0,0 +1,181 @@
+        p=0.8,
+        q=0.5
+    )
+    vertex_paths, edge_weights, vertex_path_sizes = df
+    # Check that output sizes are as expected
+    assert vertex_paths.size == max_depth * k
+    assert edge_weights.size == (max_depth - 1) * k
+    # Check that weights match up with paths
+    err = 0
+    for i in range(k):
+        for j in range(max_depth - 1):
+            # weight = edge_weights[i * (max_depth - 1) + j]
+            u = vertex_paths[i * max_depth + j]
+            v = vertex_paths[i * max_depth + j + 1]
+            # Walk not found in edgelist
+            if (not G.has_edge(u, v)):
+                err += 1
+            # FIXME: Checking weights is buggy
+            # Corresponding weight to edge is not correct
+            # expr = "(src == {} and dst == {})".format(u, v)
+            # if not (G.edgelist.edgelist_df.query(expr)["weights"] == weight):
+            #    err += 1
+    assert err == 0
+
+
+@pytest.mark.parametrize("graph_file", utils.DATASETS_SMALL)
+@pytest.mark.parametrize("directed", DIRECTED_GRAPH_OPTIONS)
+def test_node2vec_padded(
+    graph_file,
+    directed
+):
+    G = utils.generate_cugraph_graph_from_file(graph_file, directed=directed,
+                                               edgevals=True)
+    k = random.randint(1, 10)
+    max_depth = 3
+    start_vertices = random.sample(range(G.number_of_vertices()), k)
+    df, seeds = calc_node2vec(
+        G,
+        start_vertices,
+        max_depth,
+        use_padding=True,
+        p=0.8,
+        q=0.5
+    )
+    vertex_paths, edge_weights, vertex_path_sizes = df
+    # Check that output sizes are as expected
+    assert vertex_paths.size == max_depth * k
+    assert edge_weights.size == (max_depth - 1) * k
+    assert vertex_path_sizes.sum() == vertex_paths.size
+    # Check that weights match up with paths
+    err = 0
+    path_start = 0
+    for i in range(k):
+        for j in range(max_depth - 1):
+            # weight = edge_weights[i * (max_depth - 1) + j]
+            u = vertex_paths[i * max_depth + j]
+            v = vertex_paths[i * max_depth + j + 1]
+            # Walk not found in edgelist
+            if (not G.has_edge(u, v)):
+                err += 1
+            # FIXME: Checking weights is buggy
+            # Corresponding weight to edge is not correct
+            # expr = "(src == {} and dst == {})".format(u, v)
+            # if not (G.edgelist.edgelist_df.query(expr)["weights"] == weight):
+            #    err += 1
+        # Check that path sizes matches up correctly with paths
+        if vertex_paths[i * max_depth] != seeds[i]:
+            err += 1
+        path_start += vertex_path_sizes[i]
+    assert err == 0
+
+
+@pytest.mark.parametrize("graph_file", utils.DATASETS_SMALL)
+@pytest.mark.parametrize("directed", DIRECTED_GRAPH_OPTIONS)
+@pytest.mark.parametrize("max_depth", [None, -1])
+@pytest.mark.parametrize("p", [None, -1])
+@pytest.mark.parametrize("q", [None, -1])
+def test_node2vec_invalid(
```

Comment:
I modified the invalid inputs test to a series of checks instead of all possible combinations, and also removed the variability of the external parameters `graph_file` and `directed`, greatly reducing the number of tests generated but with the same coverage (more coverage bc I added strings and floats in the case of max_depth), going from 60 to 13 tests